app/javascript/controllers/ui/filter_controller.js

```diff
@@ -2,18 +2,23 @@ import { Controller } from "@hotwired/stimulus";
 
 export default class UIFilter extends Controller {
   static targets = ["source", "item"];
+  static values = {
+    mode: String,
+  };
 
   connect() {}
 
   filter(event) {
     let lowerCaseFilterTerm = this.sourceTarget.value.toLowerCase();
-    let regex = new RegExp("^" + lowerCaseFilterTerm);
```

Comment:
I have removed this since typing a "." at the search bar it was returning all the results, since . in regexp matches any character excepting line breaks.

So i replaced this with includes and startsWith.

Reply:
I propose to use the modes to allows the user to select within `plain-text` and `regex` so the user is aware of what happen if he uses a `.` (only if that user knows about regex of course)

Reply:
I like your idea, maybe to avoid the . Problem we could replace the dots with \. ? That would work I think. But the warning idea is not bad.